src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part1/index.md

```diff
@@ -0,0 +1,429 @@
+
+[![](images/Img2_WPQ_vs_mClock_Profile_Client_Pct_Lat.png)](images/Img2_WPQ_vs_mClock_Profile_Client_Pct_Lat.png)
+
+<center> Fig. 2: WPQ vs mClock Profiles - Average Client Latency and Percentile Comparison With Snaptrim Ops </center>
+
+The overall average client completion latency obtained with WPQ was 3.280 msec. However, during the snaptrim phase, the average client completion latency was 6.2 msec which is an increase of nearly 126% when compared to the baseline value of 2.747 msec!
+
+With mClock *high_client_ops* profile, the overall average completion latency was slightly higher at 3.323 msec. But during the snaptrim phase, the average completion latency was 4.506 msec which is over 27% lower than what was observed with WPQ. The table below shows the latency comparison between WPQ and mClock schedulers.
+
+| mClock Profile | Overall clat<br>(msec) | WPQ Comparison<br>(3.280 msec) | clat during<br>snaptrim(msec) | WPQ Comparison<br>(6.2 msec) |
+| :---: | :---: | :---: | :---: | :---: |
+| high_client_ops | 3.323 | 1.3% higher | 4.506 | 27% lower |
+| balanced | 3.364 | 2.6% higher | 5.074 | 18% lower |
+| high_recovery_ops | 3.506 | 6.9% higher | 5.270 | 15% lower |
+
+With the other profiles like *balanced* and *high_recovery_ops*, the overall average client completion latency increased marginally to 3.364 msec and 3.506 msec respectively. During the snaptrim phase, with the *balanced* profile, the average clat was 5.074 msec during snaptrim phase which is 18% lower than what was observed with WPQ scheduler. With the *high_recovery_ops* profile, it was 5.270 msec which is around 15% lower than what was observed with WPQ scheduler.
+
+A comparison of the average client latency variations during snaptrim ops with WPQ and mClock profiles is shown during a couple of test runs in Fig. 3 and Fig. 4. These charts track how the average client completion latency is affected during snaptrims ops.
+
+[![](images/Img3_Avg_Client_Lat_Profile_Cmp_During_Snaptrim_Ops.png)](images/Img3_Avg_Client_Lat_Profile_Cmp_During_Snaptrim_Ops.png)
+
+<center> Fig. 3: WPQ vs mClock Profiles - Average Client Latency Comparison During Snaptrim Ops </center>
+
+The latencies in both the charts can be seen to increase after 30 secs into the client I/O when snaptrim ops were triggered. Although there are intermediate latency spikes across the mClock profiles, they are predominantly lower than the WPQ latency profile.
+
+[![](images/Img4_Avg_Client_Lat_Profile_Cmp_During_Snaptrim_Ops.png)](images/Img4_Avg_Client_Lat_Profile_Cmp_During_Snaptrim_Ops.png)
+
+<center> Fig. 4: WPQ vs mClock Profiles - Average Client Latency Comparison During Snaptrim Ops </center>
+
+Therefore, it can be concluded that mClock scheduler is able to provide significantly higher  throughputs and lower latencies to client ops during snaptrim operations. But is there any impact to the snaptrim rate and duration? The following section analyzes this.
```

Comment:
```suggestion
Therefore, it can be concluded that mClock scheduler is able to provide significantly higher throughputs and lower latencies to client ops during snaptrim operations. But is there any impact to the snaptrim rate and duration? The following section analyzes this.
```

Reply:
Fixed.